tools/tpm2_createak.c

```diff
@@ -192,23 +166,55 @@ static tool_rc create_ak(ESYS_CONTEXT *ectx) {
 
     TPML_PCR_SELECTION creation_pcr = { .count = 0 };
     TPM2B_DATA outside_info = TPM2B_EMPTY_INIT;
-    TPM2B_PUBLIC *out_public;
+    TPM2B_PUBLIC *ek_public, *out_public;
     TPM2B_PRIVATE *out_private;
-    TPM2B_PUBLIC in_public = TPM2B_EMPTY_INIT;
+    TPM2B_PUBLIC in_public;
+    TPMI_ALG_HASH ek_name_alg;
+    TPML_DIGEST pHashList = { .count = 2 };
 
-    bool result = set_key_algorithm(&in_public);
-    if (!result) {
-        return tool_rc_general_error;
+    /* get the nameAlg of the EK */
+    tool_rc tmp_rc = tpm2_readpublic(ectx, ctx.ek.ek_ctx.tr_handle, &ek_public,
+            NULL, NULL);
```

Comment:
`-g` is already assigned to specify the digest algorithm used for signing. There is no special option for name halg.

 `Part 3: Commands` of the `TCG Trusted Platform Module Library` defines in section `12.1 TPM2_Create`:
> If the object being created is a Storage Key and fixedParent is SET in the attributes of inPublic, then the
symmetric algorithms and parameters of inPublic are required to match those of the parent. The
algorithms that must match are inPublic.nameAlg, ...

Actually we are creating a Signature Key so that the requirement above does not need to fulfilled. But it would make sense that the nameAlg of the Attestation Key secured be the Endorsement Key would have the same strength as the parent. Otherwise a new option for setting the name halg would be needed for `tpm2_createak`

Reply:
The nameAlg of the EK is required for the selection of the matching EK Template policy hashes, so we need the readpublic call anyway.

Reply:
> `-g` is already assigned to specify the digest algorithm used for signing. There is no special option for name halg.
> 
Ahh this is createak not ek.